main.go

```diff
@@ -52,6 +55,18 @@ func realMain(logger hclog.Logger) error {
 		return err
 	}
 
+	// Create health handler
+	mux := http.NewServeMux()
```

Comment:
Can we move this down as well so that it's closer to where it being called? It also ensures that the defer Shutdown is called first on return to signal that we're no longer healthy early on.

Reply:
Done 👍